auth.c

```diff
@@ -419,8 +419,12 @@ getpwnamallow(struct ssh *ssh, const char *user)
 	struct connection_info *ci;
 	u_int i;
 
+	if ((pw = getpwnam(user)) != NULL)
```

Comment:
you need to remove the 2nd getpwnam in line 421/435 and move the check for pw != NULL to there, no?

Reply:
what I was trying to do (badly) was avoid a situation where one of the function calls after the initial getpwnam() lookup does a getpwnam() call of its own and messes up the state. I think it's probably simplest to just do the lookup twice